src/api.ts

```diff
@@ -88,25 +91,23 @@ async function enhancedFetch<T extends string | URL>(
  * const users = await response.json(userSchema);
  * //    ^? User[]
  */
-function makeFetcher(
-  baseURL: string | URL,
-  baseHeaders?: HeadersInit | (() => HeadersInit | Promise<HeadersInit>),
-) {
+function makeFetcher(baseURL: string | URL, baseOptions: BaseOptions = {}) {
   return async <T extends string>(
     path: T,
     requestInit: EnhancedRequestInit<T> = {},
   ) => {
+    const { headers } = baseOptions
+    const requestTransformer = baseOptions.requestTransformer ?? identity
+    const responseTransformer = baseOptions.responseTransformer ?? identity
     const url = makeGetApiURL(baseURL)(path)
     const response = await enhancedFetch(url, {
-      ...requestInit,
+      ...requestTransformer(requestInit),
```

Comment:
I'm wondering if requestTransformer should also wrap the headers code below so if you are going to transform the request in any way you don't loose any transformations you end up doing in the request's headers.

Reply:
@gustavoguichard, I'm moving this discussion to #24, OK?